### PR TITLE
organize controllers-add update_hash method, refactor bitcoin integratio...

### PR DESCRIPTION
**Pull request by zhangwenx**
...n and relative tests

### CODE REVIEW

File: controllers/collections.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    """
+    Used to update collection timestamp information by new hash
+    :param collection:
+    """
+    timestamp = TimestampFile(collection.merkle).check_timestamp()
+    curr_time=time.strptime(timestamp['time'], "%Y-%m-%d %H:%M:%S")
+    curr_txs= timestamp['time'] + ";" + timestamp['Transaction']
+    edit_time (collection, curr_time, curr_txs)
+
+
```
Comment (zhangwenx) [inline controllers/collections.py]: @bmdahmen

File: controllers/collections.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          if len(string) ==0:
+              return None
+      string = string[:-1]
+      h = hashlib.sha256()
+      h.update(string)
+      collection_hash = Hash(hash=h.hexdigest(), document_ids = string, collection_ver = collection.version)
+      session = DBSession.object_session(collection)
+      session.add(collection_hash)
+      collection.hashes.append(collection_hash)
+      session.commit()
```
Comment (zhangwenx) [inline controllers/collections.py:76]: @bmdahmen still working on parameterized test for update_hash() :8ball:

**Comment by pdaian** [comment]
@zhangwenx @bmdahmen please make sure you guys add 5 unit tests for this functionality as per the iteration requirements.  Thanks! :+1:

**Comment by bmdahmen** [comment]
@pdaian  Hey, don't merge this branch. I have the same functionality in my Hash Integration branch. (I forked off @zhangwenx  's branch).

Merging Hash Integration will merge these changes in

**Comment by zhangwenx** [comment]
@bmdahmen hi, can you fork the test_update_hash.py into your branch? https://github.com/FreeJournal/freejournal/commit/0a47b6efce412ca22e63b2e6d6f2904956698b76 and https://github.com/FreeJournal/freejournal/commit/59763099d9dc9c2f34f95f90319184fb5aac59ea. then there is no differences between the two branches

**Comment by bmdahmen** [comment]
Sure, we need to put unit tests in the unittests/ folder though, I will move it but keep it in mind in the future

EDIT: after moving it I can't get the test to pass, please move it and give it another try on your pc

**Comment by zhangwenx** [comment]
@bmdahmen yes. I had the same problem but cannot fix yesterday. not sure why can not get collection from Cache after put the file into unittest/ folder.

**Pull request closed without merging**